javalib/src/main/scala/java/util/ScalaOps.scala

```diff
@@ -142,6 +150,47 @@ private[java] object ScalaOps {
       }
       result + end
     }
+
+    @inline def headOption: Option[A] = {
+      if (__self.hasNext()) Some(__self.next())
+      else None
+    }
+
+    @inline def head: A = {
+      if (__self.hasNext()) __self.next()
+      else throw new NoSuchElementException("empty.head")
+    }
+
+    @inline def lastOption: Option[A] = {
+      if (!__self.hasNext()) None
+      else {
+        var last: A = __self.next()
+        while (__self.hasNext()) {
+          last = __self.next()
+        }
+        Some(last)
+      }
+    }
+
+    @inline def last: A =
+      if (__self.hasNext()) lastOption.get
+      else throw new NoSuchElementException("empty.last")
+
+    @inline def min(comp: Comparator[_ >: A]): A =
+      reduceLeft[A] {
+        case (l, r) => if (comp.compare(l, r) <= 0) l else r
+      }
```

Comment:
There's no need for the `case`, so this can fit on one line:
```suggestion
    @inline def min(comp: Comparator[_ >: A]): A =
      reduceLeft[A]((l, r) => if (comp.compare(l, r) <= 0) l else r)
```